lib/modules/dosomething/dosomething_global/dosomething_global.module

```diff
@@ -651,6 +651,24 @@ function dosomething_global_url($path, $options = NULL) {
 }
 
 /**
+ * Build a global node URL that adds the prefix and a fragment
+ *
+ * @param string $nid
+ *    The nid of the node
+ * @param string $language
+ *    The language of the URL
+ * @param string $fragment
+ *    Optional fragment to append
+ */
+function dosomething_global_node_url($nid, $language, $fragment = NULL) {
+  $raw_url = 'node/' . $nid;
+  if (isset($fragment)) {
+    $raw_url .= '#' . $fragment;
```

Comment:
i know we've seen issue with manually concating the url like this before, it's best to pass this into url() with fragment as an option.
that will also allow for other types of urls to use this. (with a query string etc)